### PR TITLE
The complete code, tests and documentation

### PR DESCRIPTION
**Pull request by andy-maier**
Ready for review.

To review the documentation, it is best to build it locally from this branch via `make builddoc`.

### CODE REVIEW

File: tests/utils/testing_installed.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                  format(module_name, module.__file__))
+    else:
+        module = sys.modules[module_name]
+        if test_installed == 'DEBUG':
+            print("Debug: {0} module was already loaded from: {1}".
+                  format(module_name, module.__file__))
+    return module
+
+
+def is_inherited_from(member_name, derived_class, base_class):
```
Comment (andy-maier) [inline tests/utils/testing_installed.py:91]: I suggest to remove the `is_inherited_from()` this method - it is not used currently and I don't see a use forthcoming either.

Comment (andy-maier) [reply tests/utils/testing_installed.py]: DONE

File: README.rst [hunk 2]
```diff
@@ -25,54 +25,56 @@ nocaselist - A case-insensitive list for Python
...
-package into your active Python environment:
+To install the latest released version of the nocaselist package into your
+active Python environment:
 
 .. code-block:: bash
 
     $ pip install nocaselist
 
-This will also install any prerequisite Python packages.
+This would also install any prerequisite Python packages, but the nocaselist
```
Comment (KSchopmeyer) [inline README.rst:57]: Why not just say that it has no prerequisites.

File: nocaselist/_nocaselist.py [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+    result = list()
+    for value in lst:
+        result.append(value.lower())
+    return result
+
+
+class NocaseList(list):
+    # pylint: disable=line-too-long
+    """
+    A case-insensitive, but case-preserving, list.
```
Comment (KSchopmeyer) [inline nocaselist/_nocaselist.py:23]: Actually could drop the word "but"

File: nocaselist/_nocaselist.py [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+
+    The list is case-insensitive: Whenever items of the list are looked up by
+    value or item values are compared, that is done case-insensitively. The
+    case-insensitivity is defined by performing the lookup or comparison at
+    the level of the result of the ``lower()`` method on list items. Therefore,
```
Comment (KSchopmeyer) [inline nocaselist/_nocaselist.py:28]: suggest remove "at the level of" and replace 'at' with 'on'

File: README.rst (hunk 2 above)
Comment (andy-maier) [reply README.rst:57]: DONE

File: nocaselist/_nocaselist.py (hunk 3 above)
Comment (andy-maier) [reply nocaselist/_nocaselist.py:23]: DONE

File: nocaselist/_nocaselist.py (hunk 4 above)
Comment (andy-maier) [reply nocaselist/_nocaselist.py:28]: DONE

**Pull request merged**